infor-design/enterprise
# 5420 - Fix rendering of time and period in chinese language

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

This PR fixes the rendering of timepicker object `hours, minutes, and period` even if the time format is in standard time. Added a condition to identify if the period is before the time (position perse).

Basically, this fixed applies to all the standard time format `ah:mm`.

**Related github/jira issue (required)**:

Closes https://github.com/infor-design/enterprise/issues/5420

**Steps necessary to review your pull request (required)**:
1. Test this fix on the following locales
- http://localhost:4000/components/timepicker/example-index.html?locale=zh-CN
- http://localhost:4000/components/timepicker/example-index.html?locale=zh-Hans
- http://localhost:4000/components/timepicker/example-index.html?locale=zh-Hant
- http://localhost:4000/components/timepicker/example-index.html?locale=zh-TW 

```text
Follow these steps

- Pull this branch, build, and run the app
- Click the trigger button
- Set the hours, minutes, and period
- Hit the set time button
- You should not get an error message
- Open the timepicker again
- The hours, minutes and period should be the same
```

2. Optional: Run the puppeteer test to make sure it will pass
```text
npm run puppeteer -- timepicker
Should pass all the tests
```

**Included in this Pull Request**:
